### ISSUE TITLE
Ambiguity on usage of named functions

### ISSUE THREAD

**Issue by remcohaszing** [johnpapa/angular-styleguide#653]
In [Y024](https://github.com/johnpapa/angular-styleguide#style-y024) is stated that named functions should be used for callbacks.

However, in all the examples unnamed functions are passed when chaining promises. I must say using named functions in promise chaining make the control flow hard to follow from my experience.

Is Y024 only intended for defining Angular components? Because that's what the example shows.

**Comment by zachlysobey** [comment]
I agree that calling it `callbacks` is somewhat confusing.

>  Use named functions instead of passing an anonymous function in as a callback.n as a callback.

perhaps this line should instead be:

> Use named functions instead of passing an anonymous function when registering something to an angular module.

**Comment by brutalcrozt** [comment]
:+1:  agree, more
or just like this?

Use named functions instead of passing an anonymous function in as a `callback`.

**Comment by johnpapa** [comment]
but it is a callback

**Comment by johnpapa** [comment]
@remcohaszing Yes, named functions are recommended for most scenarios unless it is a 1 liner.

**Comment by zachlysobey** [comment]
To me, whether or not its a callback (which implies async unless I'm confused) under the hood isn't really something angular users _need_ to know.

Totally acceptable as is IMO however.

**Comment by dietergeerts** [comment]
Named functions are easier to follow, if a name is chosen well. Because they then say what will be done. If you don't need to go deeper, you are not bothered with it, makes it way easier to go through a flow of code imho.

**Comment by johnpapa** [comment]
The suggestions above are good ... anyone for a PR?

**Comment by johnpapa** [comment]
Thank you for the thoughtful suggestion! Since Angular 1 has reached end of life, the style guide has stabilized and we're no longer making structural changes. The guide remains a valuable reference as-is. Appreciate your contribution to the community! 🙏
